Library/Homebrew/exceptions.rb

```diff
@@ -435,11 +435,23 @@ def initialize(flags)
 # the compilers available on the user's system
 class CompilerSelectionError < RuntimeError
   def initialize(formula)
-    super <<-EOS.undent
-      #{formula.full_name} cannot be built with any available compilers.
-      To install this formula, you may need to:
-        brew install gcc
-      EOS
+    if MacOS.version > :tiger
+      super <<-EOS.undent
+        #{formula.full_name} cannot be built with any available compilers.
+        To install this formula, you may need to:
+          brew install gcc
+        EOS
```

Comment:
I had a version that tried to do this but it was really ugly. :( I ended up deciding duplicating it was better.